src/components/capture/GenerateButton.tsx

```diff
@@ -120,26 +120,21 @@ function CaptureGenerateButton({
                 });
             }
 
-            const discoversSubscription = subscription(
-                draftsResponse.data[0].id
-            );
             const discoverResponse = await discover(
                 entityName,
                 encryptedEndpointConfig.data,
                 imageConnectorTagId,
                 draftsResponse.data[0].id
             );
             if (discoverResponse.error) {
-                return callFailed(
-                    {
-                        error: {
-                            title: 'captureCreate.generate.failedErrorTitle',
-                            error: discoverResponse.error,
-                        },
+                return callFailed({
+                    error: {
+                        title: 'captureCreate.generate.failedErrorTitle',
+                        error: discoverResponse.error,
                     },
-                    discoversSubscription
-                );
+                });
             }
+            subscription(draftsResponse.data[0].id);
```

Comment:
Moving the subscription call to after the discover call is checked for errors.

Most of the "changes" here are indentation.